gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -415,10 +415,33 @@ private Entry getEntry(int affinity) {
     return localEntries.get(index);
   }
 
+  private static class RetainedEntry {
+    private final Entry entry;
+    private final AtomicBoolean wasReleased;
+
+    public RetainedEntry(Entry entry) {
+      this.entry = entry;
+      wasReleased = new AtomicBoolean(false);
+    }
+
+    void release() {
+      if (!wasReleased.compareAndSet(false, true)) {
+        Exception e = new IllegalStateException("Entry was already released");
+        LOG.log(Level.WARNING, e.getMessage(), e);
+        return;
+      }
+      entry.release();
```

Comment:
We can probably add a FINE log on success release?

Reply:
we would be generating a log message per RPC then, which would 10s of thousands per sec....it would just be noise

Reply:
Thanks! Didn't notice `release()` is called on every RPC. Can we add some comments on `release()` method?

Reply:
Added some javadoc to the class